modules/beamdyn/tests/test_tools.F90

```diff
@@ -132,7 +144,27 @@ type(BD_ParameterType) function simpleParameterType() RESULT(p)
         end do
         
     end function
-    
+   
+    subroutine SimpleParameterType_TearDown(p)
```

Comment:
It would be more accurate to call this `ParameterType_TearDown`. The "simple" part of that is only a reference to the subroutine that creates a basic instance of this data type. By the time we are cleaning up, this may be far from a "simple" `BD_ParameterType`.